com.unity.netcode.gameobjects/Runtime/Serialization/BitReader.cs

```diff
@@ -0,0 +1,217 @@
+            m_Position = m_Reader.Value.Position;
+            m_BitPosition = 0;
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+            m_AllowedBitwiseReadMark = (m_Reader.Value.AllowedReadMark - m_Position) * k_BitsPerByte;
+#endif
+        }
+
+        /// <summary>
+        /// Pads the read bit count to byte alignment and commits the read back to the reader
+        /// </summary>
+        public void Dispose()
+        {
+            var bytesWritten = m_BitPosition >> 3;
+            if (!BitAligned)
+            {
+                // Accounting for the partial read
+                ++bytesWritten;
+            }
+
+            m_Reader.Value.CommitBitwiseReads(bytesWritten);
+        }
+
+        /// <summary>
+        /// Verifies the requested bit count can be read from the buffer.
+        /// This exists as a separate method to allow multiple bit reads to be bounds checked with a single call.
+        /// If it returns false, you may not read, and in editor and development builds, attempting to do so will
+        /// throw an exception. In release builds, attempting to do so will read junk memory.
+        /// </summary>
+        /// <param name="bitCount">Number of bits you want to read, in total</param>
+        /// <returns>True if you can read, false if that would exceed buffer bounds</returns>
+        public bool TryBeginReadBits(uint bitCount)
+        {
+            var newBitPosition = m_BitPosition + bitCount;
+            var totalBytesWrittenInBitwiseContext = newBitPosition >> 3;
+            if ((newBitPosition & 7) != 0)
+            {
+                // Accounting for the partial read
+                ++totalBytesWrittenInBitwiseContext;
+            }
+
+            if (m_Reader.Value.PositionInternal + totalBytesWrittenInBitwiseContext > m_Reader.Value.LengthInternal)
+            {
+                return false;
+            }
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+            m_AllowedBitwiseReadMark = (int)newBitPosition;
+#endif
+            return true;
+        }
+
+        /// <summary>
+        /// Read a certain amount of bits from the stream.
+        /// </summary>
+        /// <param name="value">Value to store bits into.</param>
+        /// <param name="bitCount">Amount of bits to read</param>
+        public unsafe void ReadBits(out ulong value, uint bitCount)
+        {
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+            if (bitCount > 64)
+            {
+                throw new ArgumentOutOfRangeException(nameof(bitCount), "Cannot read more than 64 bits from a 64-bit value!");
+            }
+
+            if (bitCount < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(bitCount), "Cannot read fewer than 0 bits!");
+            }
+
+            int checkPos = (int)(m_BitPosition + bitCount);
+            if (checkPos > m_AllowedBitwiseReadMark)
+            {
+                throw new OverflowException("Attempted to read without first calling TryBeginReadBits()");
```

Comment:
a minor suggestion, use `nameof()` in conjunction with with string-interpolation:
```suggestion
                throw new OverflowException($"Attempted to read without first calling {nameof(TryBeginReadBits)}()");
```

Reply:
Done